CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,107 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+
+
+# Function for opening the file
+def CAN_Decoder():
+    lineNum = 0
+    oldDataFile = filedialog.askopenfile(initialdir="/")
+    listedOldData = []
+    for dataLine in oldDataFile:
+        segmentedDataLine = dataLine.split(" ")
+        listedOldDataLine = []
+        for stringSegment in segmentedDataLine:
+            if stringSegment.strip():
+                listedOldDataLine.append(stringSegment)
+        listedOldData.append(listedOldDataLine)
+        lineNum = lineNum + 1
+
+    newData = []
+    lineCount = len(listedOldData)
+    for dataLineIndex in range(0, lineCount):
+        dataLine = listedOldData[dataLineIndex]
+        refinedDataLine = []
+        frameType = int(dataLine[3])
```

Comment:
it's better to transform all data to int before start the ISO-TP reassembly

Reply:
All payload data are not convertible to int  for e.g. 1C, 6C, etc.  Please suggest